### PR TITLE
Better sampling of sigma_8

### PR DESCRIPTION
**Pull request by tilmantroester**
This PR replaces the janky sigma_8 sampling setup. Instead of running CAMB twice, the transfer function (which takes the most time) is only computed once and then the power spectra are recomputed using the correct A_s corresponding to the requested sigma_8.

This speeds things up by ~50% and also avoids the numerical uncertainty in the initial estimate of sigma_8 due to the low accuracy settings of the first run. Since `sigma_8_input` now agrees with the output `sigma_8`, I added the option of taking `sigma_8` as the input directly. If both `sigma_8` and `A_s` are provided, `sigma_8` takes precedence.

This approach could in principle be used to do fast sampling of all non-cosmology parameters, such as primordial power spectrum parameters in general or HMCode parameters. See for example cells 50 onward here: https://camb.readthedocs.io/en/latest/CAMBdemo.html

### CODE REVIEW

**Review by joezuntz — CHANGES_REQUESTED** [review]
One more request - could you update e.g the des-y1 ini file to use the new name. That is tested in the CI.

File: boltzmann/camb/camb_interface.py [hunk 1]
```diff
@@ -649,6 +625,10 @@ def execute(block, config):
         else:
             # other modes
             r = camb.get_results(p)
+            if (block.has_value(cosmo, 'sigma_8_input')
```
[the remark this answers is not in the document]
Comment (tilmantroester) [reply boltzmann/camb/camb_interface.py:628]: So no backwards compatibility?

[the remark this answers is not in the document]
Comment (joezuntz) [reply boltzmann/camb/camb_interface.py:628]: Good point. We should drop one of them, I guess it's clarity vs compatibility.  You're right, let's stick with compatibility and disallow sigma_8.

[the remark this answers is not in the document]
Comment (tilmantroester) [reply boltzmann/camb/camb_interface.py]: Added a deprecation warning for sigma_8_input but allowing both for now.

**Comment by tilmantroester** [comment]
Back to you!

**Comment by joezuntz** [comment]
Brilliant, thanks for this!

**Pull request merged**